web3/contract.py

```diff
@@ -1323,14 +1344,17 @@ def call_contract_function(abi,
         fn_args=args,
         fn_kwargs=kwargs,
         transaction=transaction,
+        is_function_abi=is_function_abi,
     )
 
     if block_id is None:
         return_data = web3.eth.call(call_transaction)
     else:
         return_data = web3.eth.call(call_transaction, block_identifier=block_id)
 
-    function_abi = find_matching_fn_abi(abi, function_identifier, args, kwargs)
+    function_abi = abi
+    if not is_function_abi:
+        function_abi = find_matching_fn_abi(abi, function_identifier, args, kwargs)
```

Comment:
This shouldn't block the PR, but next time you run into something like it, consider this pattern:

```py
    if is_function_abi:
        function_abi = abi
    else:
        function_abi = find_matching_fn_abi(abi, function_identifier, args, kwargs)
```

Piper convinced me that this style of "set a variable a different way in each conditional branch" is easier to read, and less error prone (when maintaining the code over time) than the pattern of "set the variable and re-set it depending on a conditional".

Reply:
totally agree. Looks much cleaner